### PR TITLE
docs: add documentation styleguide

### PR DESCRIPTION
**Pull request by sianford**
### Proposed behaviour
Create a Documentation styleguide to aid with creating and maintaining documents throughout the repository with a consistent style.

### Current behaviour
There is no Documentation styleguide for contributors to follow.

### Checklist

<del>- [ ] Screenshots are included in the PR
<del>- [ ] Carbon implementation and Design System documentation are congruent
<del>- [ ] All themes are supported
- [ ] Commits follow our style guide

<del>- [ ] Unit tests added or updated
<del>- [ ] Cypress automation tests added or updated
<del>- [ ] Storybook added or updated
<del>- [ ] Typescript `d.ts` file added or updated

### Testing instructions
Check links work in the document.

### CODE REVIEW

File: docs/documentation-styleguide.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+[Use of Links](#use-of-links)
+* [Linking to other sites](#linking-to-other-sites)
+* [Linking to other documents](#linking-to-other-documents)
+
+[Use of Images](#use-of-images)
+
+
+### Introduction
+This Documentation styleguide describes what is documented in the Carbon repository and which format this  takes to enable you to contribute documentation in line with the current styles. It includes:
```
Comment (DlgSHi) [inline docs/documentation-styleguide.md:39]:
```suggestion
This Documentation style guide describes what is documented in the Carbon repository and which format this takes to enable you to contribute documentation in line with the current styles. It includes:
```

File: docs/documentation-styleguide.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+- where documents live within the repository;
+- which format is used for which purpose;
+- language and grammar style;
+- linking to other documents and sites; and
```
Comment (DlgSHi) [inline docs/documentation-styleguide.md:43]:
```suggestion
- linking to other documents and sites;
```

File: docs/documentation-styleguide.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+We write README files in MD format to tell collaborators about what Carbon and [Cypress.io](https://www.cypress.io) are, how collaborators can get started and where they can find further information or assistance.
+
+The [Carbon README](../README.md) contains:
+- a link to a more detailed 'Getting Started' guide;
+- a link to the [contributing guide](../CONTRIBUTING.md) explaing how to contribute to the project; and
+- license information.
+
+The [Cypress README](../cypress/README.md) contains:
+- instructions of how to install and run Cypress;
+- how to access test results; and
```
Comment (DlgSHi) [inline docs/documentation-styleguide.md:78]:
```suggestion
- how to access test results;
```

File: docs/documentation-styleguide.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+The [RFCs README](../rfcs/README.md) provides detailed instructions of when and how to write an RFC. 
+
+All README files are accessible through GitHub from their respective directories.
+
+When writing a README file, consider:
+- who your audience is;
+- what useful information you want in there;
+- if a table of contents is required;
+- the [use of links](#use-of-links); and
```
Comment (DlgSHi) [inline docs/documentation-styleguide.md:89]:
```suggestion
- the [use of links](#use-of-links);
```

File: docs/documentation-styleguide.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+#### Styleguides
+We write various styleguides to provide some consistency for collaborators to the Carbon project to follow. This helps to maintain a standard way of doing things so we have a limited set of methods, tools and styles adopted throughout the repository. They often contain ordered and concise instructions so users can follow and repeat them easily. This is particularly useful for people new to the project who seek some common guidelines.
+
+Styleguides are stored in the `/docs` directory. It is a living directory so it continuously changes, with new guides added and others amended or removed. 
+
+When writing a styleguide, consider:
+- who your audience is and what prior understanding they have;
+- if a glossary is required;
+- formatting of [code in text](#code-in-text);
+- the [use of links](#use-of-links); and
```
Comment (DlgSHi) [inline docs/documentation-styleguide.md:106]:
```suggestion
- the [use of links](#use-of-links);
```

File: docs/documentation-styleguide.md (hunk 2 above)
Comment (sianford) [reply docs/documentation-styleguide.md:43]: I followed the first example in "A list of very short points" in this guide for bullet point lists: http://www.plainenglish.co.uk/punctuating-bulleted-lists.html. However, if you think the second example without an "and" before the last point, then I'm happy to change.

File: docs/documentation-styleguide.md (hunk 3 above)
Comment (sianford) [reply docs/documentation-styleguide.md:78]: I followed the first example in "A list of very short points" in this guide for bullet point lists: http://www.plainenglish.co.uk/punctuating-bulleted-lists.html. However, if you think the second example without an "and" before the last point, then I'm happy to change.

File: docs/documentation-styleguide.md (hunk 4 above)
Comment (sianford) [reply docs/documentation-styleguide.md:89]: I followed the first example in "A list of very short points" in this guide for bullet point lists: http://www.plainenglish.co.uk/punctuating-bulleted-lists.html. However, if you think the second example without an "and" before the last point, then I'm happy to change.

File: docs/documentation-styleguide.md (hunk 5 above)
Comment (sianford) [reply docs/documentation-styleguide.md:106]: I followed the first example in "A list of very short points" in this guide for bullet point lists: http://www.plainenglish.co.uk/punctuating-bulleted-lists.html. However, if you think the second example without an "and" before the last point, then I'm happy to change.

File: docs/documentation-styleguide.md (hunk 1 above)
Comment (sianford) [reply docs/documentation-styleguide.md:39]: I couldn't see that extra space between "this" and "takes" as the word "takes" had wrapped onto the next line in VS Code. Thanks 👍

Comment (sianford) [reply docs/documentation-styleguide.md]: Separated "styleguide" into two words "style guide" throughout, apart from where it quotes another document.

File: docs/documentation-styleguide.md (hunk 2 above)
Comment (sianford) [reply docs/documentation-styleguide.md]: Removed the "and" in all lists in this document.

File: docs/documentation-styleguide.md (hunk 3 above)
Comment (sianford) [reply docs/documentation-styleguide.md]: Removed the "and" in all lists in this document.

File: docs/documentation-styleguide.md (hunk 4 above)
Comment (sianford) [reply docs/documentation-styleguide.md]: Removed the "and" in all lists in this document.

File: docs/documentation-styleguide.md (hunk 5 above)
Comment (sianford) [reply docs/documentation-styleguide.md]: Removed the "and" in all lists in this document.

File: docs/documentation-styleguide.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+When you want to propose a substantial change to Carbon, such as the addition of a new component or a breaking change, you need to submit an RFC. The core Carbon Team will discuss the RFC needs to incorporate design constraints and reach a consensus before it is progressed.
+
+There is a [template](../rfcs/template.md) to guide you when writing an RFC. You can add other sections if appropriate. Use hyphens to separate words in the file name and save it as a `.md` file in the `carbon/rfcs/text` directory. If you need to add supporting images, save them in the `carbon/rfcs/images` directory.
+
+#### README files
+We write README files in MD format to tell collaborators about what Carbon and [Cypress.io](https://www.cypress.io) are, how collaborators can get started and where they can find further information or assistance.
+
+The [Carbon README](../README.md) contains:
+- a link to a more detailed 'Getting Started' guide;
+- a link to the [contributing guide](../CONTRIBUTING.md) explaing how to contribute to the project; and
```
Comment (DlgSHi) [inline docs/documentation-styleguide.md:73]: one more `and` missed :(

Comment (sianford) [reply docs/documentation-styleguide.md]: 🤦‍♀️ Corrected.

File: docs/documentation-styleguide.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+## Follow these guidelines when contributing documentation to the Carbon repository
+
+### Contents
+
+[Introduction](#introduction)
+
+[What We Document](#what-we-document)
+* [User Facing Documentation](#user-facing-documentation)
+* [Release Notes and Changelog](#release-notes-and-changelog)
+* [Request For Change (RFC)](#request-for-change)
```
Comment (ianforster-sage) [inline docs/documentation-styleguide.md:11]: Could you double check that this RFC skip-link works?

Comment (sianford) [reply docs/documentation-styleguide.md:11]: I missed off "-rfc" 😞 I added that afterwards and didn't update the link. Corrected.

**Pull request merged**